modules/skybrowser/src/wwtdatahandler.cpp

```diff
@@ -0,0 +1,366 @@
+#include <modules/skybrowser/include/wwtdatahandler.h>
+
+#include <modules/skybrowser/include/utility.h>
+#include <modules/space/speckloader.h>
+#include <openspace/util/httprequest.h> // For downloading files from url
+#include <ghoul/filesystem/filesystem.h>
+#include <ghoul/logging/logmanager.h>
+#include <filesystem> // To iterate through files in directory
+#include <algorithm>
+#include <sys/types.h>
+#include <sys/stat.h>
+
+namespace {
+    constexpr const char* _loggerCat = "WwtDataHandler";
+} //namespace
+
+namespace openspace {
+
+    bool hasAttribute(const tinyxml2::XMLElement* element, const std::string& name) {
+         return element->FindAttribute(name.c_str());
+    }
+
+    std::string attribute(const tinyxml2::XMLElement* element, const std::string& name) {
+        if (hasAttribute(element, name)) {
+            return element->FindAttribute(name.c_str())->Value();
+        }
+        else {
+            return wwt::Undefined;
+        }
+    }
+
+    // Parsing and downloading of wtml files
+    bool downloadFile(const std::string& url, const std::string& fileDestination) {
+        // Get the web page and save to file
+        HttpFileDownload wtml_root(
+            url, fileDestination, HttpFileDownload::Overwrite::Yes
+        );
+        wtml_root.start(std::chrono::milliseconds(5));
+        return wtml_root.hasSucceeded();
+    }
+
+    bool directoryExists(const std::string& path)
+    {
+        struct stat info;
+
+        int statRC = stat(path.c_str(), &info);
```

Comment:
These functions can be replaced with std::filesystem calls:
basically:  `const bool directoryExists = std::filesystem::exists(path) && std::filesystem::is_drectory(path);`